int32 new_heading
---
